src/view/newtListMenu.cpp

```diff
@@ -6,3 +6,76 @@
 // This file is empty since the method is now a template
 
 #include <cloysterhpc/view/newt.h>
+#include <newt.h>
+#include <ranges>
+
+static std::vector<std::string> retrieveListSelectedItems(newtComponent list)
+{
+    int selectCount = 0;
+    char** selectItems
+        = reinterpret_cast<char**>(newtListboxGetSelection(list, &selectCount));
+    auto temp = std::vector(selectItems, selectItems + selectCount)
+        | std::views::transform([](char* c) { return std::string { c }; });
+    std::vector<std::string> ret(std::begin(temp), std::end(temp));
+    return ret;
+}
+
+std::pair<int, std::vector<std::string>> Newt::multipleSelectionMenu(
+    const char* title, const char* message, const char* help,
+    std::vector<std::tuple<std::string, std::string, bool>> items)
+{
+    auto* form = newtForm(nullptr, nullptr, 0);
+    newtFormSetWidth(form, m_suggestedWidth);
+
+    auto* label = newtLabel(1, 1, message);
+    auto stitle = std::string { title };
+
+    auto* list = newtListbox(1, 3, m_maxListHeight, NEWT_FLAG_MULTIPLE);
+
+    for (const auto& [key, item, enabled] : items) {
+        newtListboxAppendEntry(list, item.c_str(), (void*)key.c_str());
+
+        if (enabled)
+            newtListboxSelectItem(list, key.c_str(), NEWT_FLAGS_SET);
+    }
+
+    newtGrid grid = newtCreateGrid(1, 3);
+    newtComponent bOk, bCancel, bHelp;
+    newtGrid buttonGrid = newtButtonBar(const_cast<char*>(TUIText::Buttons::ok),
+        &bOk, const_cast<char*>(TUIText::Buttons::cancel), &bCancel,
+        const_cast<char*>(TUIText::Buttons::help), &bHelp, NULL);
+    newtGridSetField(grid, 0, 0, NEWT_GRID_COMPONENT, label, 1, 1, 0, 0, 0,
+        NEWT_GRID_FLAG_GROWX);
+    newtGridSetField(grid, 0, 1, NEWT_GRID_COMPONENT, list, 1, 1, 0, 0, 0,
+        NEWT_GRID_FLAG_GROWX | NEWT_GRID_FLAG_GROWY);
+    newtGridSetField(grid, 0, 2, NEWT_GRID_SUBGRID, buttonGrid, 0, 1, 0, 0, 0,
+        NEWT_GRID_FLAG_GROWX);
+    newtGridWrappedWindow(grid, const_cast<char*>(stitle.c_str()));
+
+    newtFormAddComponents(form, list, label, bOk, bCancel, bHelp, nullptr);
+
+    newtExitStruct es = {};
+    newtFormRun(form, &es);
+
+    int retval = 0;
+
+    while (retval == 0) {
+        while (es.reason == 2) {
+            newtFormRun(form, &es);
+        }
+
+        if (es.u.co == bOk) {
+            retval = 1;
+        } else if (es.u.co == bHelp) {
+            this->helpMessage(help);
+        } else {
+            retval = 2;
+        }
```

Comment:
`switch`?

Also is this the handling of the buttons? If yes we have a standard way to handle it: https://github.com/viniciusferrao/cloysterhpc/blob/ed75e5d589cab8b9c3e4a0dd6f16d49c2ceda4e4/src/view/newtYesNoQuestion.cpp#L29

However I think that should be another method, to avoid code duplication. Not sure how to achieve that on the context.

Reply:
> `switch?`

The, `switch` only would work here if the `bOk` and `bHelp` were constexpr; and I don't think this is applicable, because the button creation function is not constexpr

> Also is this the handling of the buttons? If yes we have a standard way to handle it:

Yes, but since this is a custom-built form (because we needed to select multiple repos, something that the default list form does not allow us to do), we do not have the error code treatment that the easy form functions Newt give to us

> However I think that should be another method, to avoid code duplication. Not sure how to achieve that on the context.

I agree, but I need to think a little bit more about that

Reply:
Ok @arthurmco if you didn't came up with a solution in a timely manner I think we can accept this. I was wondering for a total overhaul of the TUI in the future.

However here, at least if you don't mind using a better naming in the variables, like `button` instead of `b`. Is good. To be hones is a little bad to read `es.u.co == b0k`.

Reply:
Ok. Did it in the last commit